src/main/java/TaskList.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * This is the abstract class TaskList. This class is used to add and remove tasks from the current list.
+ * @author Hua Lun
+ */
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+
+abstract class TaskList {
+    private int taskNumber;
+    private String taskCheck;
+    private String taskName;
+    private String type;
+
+    public TaskList(int i, String tC, String tN, String t) {
+        taskNumber = i;
+        taskCheck = tC;
+        taskName = tN;
+        type = t;
+    }
+
+
+    public abstract DateTime getAB();
+
+    /**
+     * <p>
+     *     getTaskNumber is used to retrieve the task number
+     * </p>
+     * @return the task number
+     */
+
+
+    public int getTaskNumber() {
+        return taskNumber;
+    }
+
+
+    /**
+     * <p>
+     *     getTaskCheck is used to check if the task is done or not
+     * </p>
+     * @return status of the task
+     */
+
+    public String getTaskCheck() {
+        return taskCheck;
+    }
+
+    /**
+     * <p>
+     *     getTaskName is used to retrieve the name of the task
+     * </p>
+     * @return task's name
+     */
+
+
+    public String getTaskName() {
+        return taskName;
+    }
+
+  
+    /**
+     * <p>
+     *     getTyoe is used to retrieve the type of task given
+     * </p>
+     * @return task's type
+     */
+
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * <p>
+     *     changeTaskCheck is used to mark the task as complete
+     * </p>
+     */
+
+
+    public void changeTaskCheck() {
+        taskCheck = "[✓]";
+    }
+
+    /**
+     * <p>
+     *     printList is used to print out all the task in the list
+     * </p>
+     * @param a the list of task
+     */
+
+    static void printList(ArrayList<TaskList> a) {
+        System.out.println("Here are the tasks in your list:");
+        for (TaskList t : a) {
+            System.out.println(t);
+        }
+    }
+
+
+    /**
+     * <p>
+     *     markAsDone is used to mark the given task as completed
+     * </p>
+     * @param i the completed task's number
+     * @param a the list of task
+     */
+
+    static void markAsDone(int i, ArrayList<TaskList> a) {
+        System.out.println("Nice! I've marked this task as done:");
+        TaskList t = a.get(i - 1);
+        String currentTask = t.getTaskName();
+        String getType = t.getType();
+        TaskList doneTask;
+        if(getType.equals("todo")) {
```

Comment:
use the change checkmark function, also why do you have to take it out and create a new one?